integreat_cms/cms/views/push_notifications/push_notification_form_view.py

```diff
@@ -344,6 +286,96 @@ def post(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
         )
 
 
+# pylint: disable=too-many-branches
+def validate_and_save_forms(
+    request: HttpRequest,
+    details: dict,
+    instance: PushNotification,
+    pn_form: PushNotificationForm,
+    pnt_formset: Any,
+) -> bool:
+    """
+    Validates and saves the forms and returns `True` iff not errors occurred.
+    :param request: The request
+    :param details: The push notification details
+    :param instance: The push notification instance
+    :param pn_form: The push notification form
+    :param pnt_formset: The push notification translation formset
+    :return: whether verification was successful
+    """
+    if details["disable_edit"]:
+        not_accessible_regions_warning = __(
+            _(
+                "This news is also assigned to regions you don't have access to: {}."
+            ).format(", ".join(map(str, details["other_regions"]))),
+            _("Thus you cannot edit or delete this news."),
+            _(
+                "Please ask the one responsible for all of these regions or contact an administrator."
+            ),
+        )
+        messages.error(
```

Comment:
From our Barcamp on Clean Code I remember that messages should be separated from the logic as they are on different levels. In addition to this I think this function is again a bit too long and does too many things. Do you think it makes sense to split this up a little bit?

Reply:
I agree that it makes sense to split up the `validate_and_save_forms` function, and i pushed a commit with a separate function for validation and saving. I am not convinced that we should changed the way messages are created, because I think that it would increase complexity, rather than lower for those concrete functions. Or do you have a specific implementation of that in mind?

Reply:
Fair enough. I like it better like this already :)